kratos/tests/test_mortar_mapper.py

```diff
@@ -0,0 +1,187 @@
+        KratosMultiphysics.VariableUtils().SetFlag(KratosMultiphysics.MASTER, False, self.model_part_slave.Nodes)
+
+        self.model_part_master = self.main_model_part.GetSubModelPart("Parts_Parts_Auto2")
+        KratosMultiphysics.VariableUtils().SetFlag(KratosMultiphysics.SLAVE, False, self.model_part_master.Conditions)
+        KratosMultiphysics.VariableUtils().SetFlag(KratosMultiphysics.MASTER, True, self.model_part_master.Conditions)
+        KratosMultiphysics.VariableUtils().SetFlag(KratosMultiphysics.SLAVE, False, self.model_part_master.Nodes)
+        KratosMultiphysics.VariableUtils().SetFlag(KratosMultiphysics.MASTER, True, self.model_part_master.Nodes)
+
+        for node in self.model_part_master.Nodes:
+            x = node.X
+            y = node.Y
+            z = node.Z
+            node.SetSolutionStepValue(KratosMultiphysics.TEMPERATURE, z)
+            node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X, x)
+            node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y, y)
+            node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Z, z)
+        del(node)
+
+        map_parameters = KratosMultiphysics.Parameters("""
+        {
+            "echo_level"                       : 0,
+            "absolute_convergence_tolerance"   : 1.0e-9,
+            "relative_convergence_tolerance"   : 1.0e-4,
+            "max_number_iterations"            : 10,
+            "integration_order"                : 2
+        }
+        """)
+
+        if (pure_implicit == True):
+            #linear_solver = ExternalSolversApplication.SuperLUSolver()
+            linear_solver = KratosMultiphysics.SkylineLUFactorizationSolver()
+
+            if (num_nodes == 3):
+                self.mortar_mapping_double = KratosMultiphysics.SimpleMortarMapperProcess3D3NDoubleHistorical(self.main_model_part, KratosMultiphysics.TEMPERATURE, map_parameters, linear_solver)
+                self.mortar_mapping_vector = KratosMultiphysics.SimpleMortarMapperProcess3D3NVectorHistorical(self.main_model_part, KratosMultiphysics.DISPLACEMENT, map_parameters, linear_solver)
+            else:
+                self.mortar_mapping_double = KratosMultiphysics.SimpleMortarMapperProcess3D4NDoubleHistorical(self.main_model_part, KratosMultiphysics.TEMPERATURE, map_parameters, linear_solver)
+                self.mortar_mapping_vector = KratosMultiphysics.SimpleMortarMapperProcess3D4NVectorHistorical(self.main_model_part, KratosMultiphysics.DISPLACEMENT, map_parameters, linear_solver)
+        else:
+            if (num_nodes == 3):
+                self.mortar_mapping_double = KratosMultiphysics.SimpleMortarMapperProcess3D3NDoubleHistorical(self.main_model_part, KratosMultiphysics.TEMPERATURE, map_parameters)
+                self.mortar_mapping_vector = KratosMultiphysics.SimpleMortarMapperProcess3D3NVectorHistorical(self.main_model_part, KratosMultiphysics.DISPLACEMENT, map_parameters)
```

Comment:
maybe you could template the scalar and vector version?

Also one thing could be for Historical/NonHistorical to use `Kratos::Flags`
(I call the mapping function with mapper.Map(OriginVariable, DestinationVariable, Kratos::Flags), where the flags can be ADD_VALUES, SWAP_SIGN,... )

Reply:
The this is that this is a process, not an utility, so the execute performs the algorithm, but the variables cannot be set outside the constructor. I can think about the flags